serverjs/render.js

```diff
@@ -0,0 +1,79 @@
+  pages.CubeComparePage = require('../dist/pages/CubeComparePage').default;
+  pages.CubeAnalysisPage = require('../dist/pages/CubeAnalysisPage').default;
+  pages.CubeBlogPage = require('../dist/pages/CubeBlogPage').default;
+  pages.CubeDeckPage = require('../dist/pages/CubeDeckPage').default;
+  pages.CubeDeckbuilderPage = require('../dist/pages/CubeDeckbuilderPage').default;
+  pages.CubeDecksPage = require('../dist/pages/CubeDecksPage').default;
+  pages.CubeDraftPage = require('../dist/pages/CubeDraftPage').default;
+  pages.CubeListPage = require('../dist/pages/CubeListPage').default;
+  pages.CubeOverviewPage = require('../dist/pages/CubeOverviewPage').default;
+  pages.CubePlaytestPage = require('../dist/pages/CubePlaytestPage').default;
+  pages.CubeSamplePackPage = require('../dist/pages/CubeSamplePackPage').default;
+  pages.CubeGridDraftPage = require('../dist/pages/GridDraftPage').default;
+  pages.ContactPage = require('../dist/pages/ContactPage').default;
+  pages.InfoPage = require('../dist/pages/InfoPage').default;
+  pages.DonatePage = require('../dist/pages/DonatePage').default;
+  pages.DownTimePage = require('../dist/pages/DownTimePage').default;
+  pages.FiltersPage = require('../dist/pages/FiltersPage').default;
+  pages.CardSearchPage = require('../dist/pages/CardSearchPage').default;
+  pages.TopCardsPage = require('../dist/pages/TopCardsPage').default;
+  pages.CardPage = require('../dist/pages/CardPage').default;
+  pages.CommentPage = require('../dist/pages/CommentPage').default;
+  pages.LoginPage = require('../dist/pages/LoginPage').default;
+  pages.RegisterPage = require('../dist/pages/RegisterPage').default;
+  pages.LostPasswordPage = require('../dist/pages/LostPasswordPage').default;
+  pages.NotificationsPage = require('../dist/pages/NotificationsPage').default;
+  pages.PasswordResetPage = require('../dist/pages/PasswordResetPage').default;
+  pages.UserAccountPage = require('../dist/pages/UserAccountPage').default;
+  pages.UserBlogPage = require('../dist/pages/UserBlogPage').default;
+  pages.UserDecksPage = require('../dist/pages/UserDecksPage').default;
+  pages.UserSocialPage = require('../dist/pages/UserSocialPage').default;
+  pages.UserCubePage = require('../dist/pages/UserCubePage').default;
+}
+
+const getPage = (page) => pages[page] || pages.Loading;
+
+const render = (req, res, page, reactProps = {}, options = {}) => {
+  reactProps.user = req.user
+    ? {
+        id: req.user._id,
+        notifications: req.user.notifications,
+        username: req.user.username,
+        email: req.user.email,
+        about: req.user.about,
+        image: req.user.image,
+        image_name: req.user.image_name,
+        artist: req.user.artist,
+      }
+    : null;
+
+  return res.render('main', {
+    reactHTML:
+      NODE_ENV === 'production' ? ReactDOMServer.renderToString(React.createElement(getPage(page), reactProps)) : null,
```

Comment:
Moving every page to SSR will increase server load. It's probably fine, but we should just be aware. Now that we're doing full react, we might be able to switch to `renderToNodeStream` which will be much more performant. I can think about how to do that if you want. It is possible to move the HTML header into react too by using stuff like Helmet.

Reply:
Server CPU is basically never the bottleneck, so I think this will be alright.